packages-exp/auth-types-exp/index.d.ts

```diff
@@ -386,9 +388,9 @@ export interface User extends UserInfo {
 /**
  * https://firebase.google.com/docs/reference/js/firebase.auth#usercredential
  */
-export interface UserCredential {
+export interface UserCredential<T extends AuthCredential | null> {
```

Comment:
@samtstern This whole PR for this one little change.

This will (eventually) address the issue you raised the other day

Reply:
should we run this change past a larger audience? seem like a pretty big signature change to me...